ledger-core-ethereum/test/SynchronizationTests.cpp

```diff
@@ -0,0 +1,464 @@
+                    EXPECT_EQ(erc20BalancesFromAccount[0]->toString(10), erc20Balance->toString(10));
+
+                    auto amountToSend = std::make_shared<BigInt>(BigInt::fromString("10"));
+                    auto transferData = wait(std::dynamic_pointer_cast<ERC20LikeAccount>(erc20Accounts[0])->getTransferToAddressData(amountToSend, "0xabf06640f8ca8fC5e0Ed471b10BeFCDf65A33e43"));
+                    EXPECT_GT(transferData.size(), 0);
+
+                    auto operations = wait(std::dynamic_pointer_cast<ERC20LikeOperationQuery>(erc20Accounts[0]
+                            ->queryOperations()
+                            ->addOrder(api::OperationOrderKey::DATE, true)
+                            ->complete())->execute());
+                    std::cout << "ERC20 Operations: " << operations.size() << std::endl;
+                    EXPECT_NE(operations.size(), 0);
+
+                    int64_t gasPrice = 20, gasLimit = 200;
+                    auto currency = account->getWallet()->getCurrency();
+                    auto txBuilder = std::dynamic_pointer_cast<EthereumLikeTransactionBuilder>(account->buildTransaction());
+                    txBuilder->setGasPrice(api::Amount::fromLong(currency, 20));
+                    txBuilder->setGasLimit(api::Amount::fromLong(currency, 200));
+                    txBuilder->wipeToAddress("0xfb98bdd04d82648f25e67041d6e27a866bec0b47");
+                    auto tx = wait(txBuilder->build());
+                    EXPECT_EQ(tx->getValue()->toLong(), balance->toLong() - (gasLimit * gasPrice));
+
+                    dispatcher->stop();
+                });
+
+                auto restoreKey = account->getRestoreKey();
+                account->synchronize()->subscribe(dispatcher->getMainExecutionContext(),receiver);
+
+                dispatcher->waitUntilStopped();
+
+                auto ops = wait(std::dynamic_pointer_cast<EthereumLikeOperationQuery>(account->queryOperations()->complete())->execute());
+                std::cout << "Ops: " << ops.size() << std::endl;
+
+                auto block = wait(account->getLastBlock());
+                auto blockHash = block.blockHash;
+            }
+        }
+    }
+    // Recover account
+    {
+        auto services = newDefaultServices();
+
+        auto walletStore = newWalletStore(services);
+        walletStore->addCurrency(ledger::core::currencies::ethereum());
+
+        auto factory = std::make_shared<EthereumLikeWalletFactory>(ledger::core::currencies::ethereum(), services);
+        walletStore->registerFactory(ledger::core::currencies::ethereum(), factory);
+
+        auto wallet = wait(walletStore->getWallet(walletName));
+        auto account = std::dynamic_pointer_cast<EthereumLikeAccount>(wait(wallet->getAccount(0)));
+        EXPECT_EQ(account->getERC20Accounts().size(), erc20Count);
+
+        wait(services->freshResetAll());
+    }
+}
+
+//TEST_F(EthereumLikeWalletSynchronization, BalanceHistory) {
```

Comment:
Why all the below tests are commented ?

Reply:
Because I’ve forgot about them. Dammit. Good catch :D :D :D

Reply:
_c’était pour voir si tu suivais_ 😆